src/sentry/eventstream/kafka/protocol.py

```diff
@@ -172,9 +190,27 @@ def get_task_kwargs_for_message_from_headers(
                 "is_regression": is_regression,
                 "is_new_group_environment": is_new_group_environment,
             }
+
+            if "group_states" not in header_data:
+                header_data["group_states"] = None
+
+            group_states_str = None
+            try:
+                group_states_str = decode_optional_str(header_data["group_states"])
```

Comment:
Does it makes sense to use `decode_optional_str(header_data.get("group_states"))` and skip the earlier check and mutating the header data?